components/InputScanner.js

```diff
@@ -1 +1,58 @@
-// hhjg
+import React, { Component } from 'react';
+import { Text, View, StyleSheet, Alert } from 'react-native';
+import { Constants, BarCodeScanner, Permissions } from 'expo';
+
+class InputScanner extends React.Component {
+  state = {
+    hasCameraPermission: null,
+    scannedBarCode: null,
```

Comment:
Nice - again that can be access with InputScanner.state.scannedBarCode. 
We'll need to have a way to check whether the code comes from the manual input or scanned input.